daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/Assert.scala

```diff
@@ -91,6 +91,12 @@ object Assert extends Assert {
   def usage(testAbortsIfFalse: Boolean, message: String): Unit = macro AssertMacros.usageMacro2
   def usage(testAbortsIfFalse: Boolean): Unit = macro AssertMacros.usageMacro1
 
+  /**
+   * Test for the validity of the arguments of the API functions
+   */
+  def argCheck(testThrowsIfFalse: Boolean, message: String): Unit =
```

Comment:
Do we want a name other than "argCheck" for checking public API stuff? The isError check isn't checking an argument, it's just checking internal state.

Reply:
Does this mean for argCheck we should be throwing IllegalArgumentException?

Reply:
Yeah, an argCheck function probably wants to either call abort or throw an IllegalArgumentException. But our uses aren't really about checking args, so I'd suggest we rename it. See other comment.